### PR TITLE
chore(tests): reuse users throughout test suite

### PR DESCRIPTION
**Pull request by genehynson**
The `ui-test-provisioner` has been updated to provide the ability to provision and flush individual users by passing them into the query params like `?user=23mcfly@influxdata.com`. This PR uses this ability so we can reuse users across all the tests in a single test suite. This will keep us from running into the "ran out of 1250 provisioned users" issue

### CODE REVIEW

**Review by TCL735 — APPROVED** [review]

**Review by joelfried — APPROVED** [review]

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -575,16 +575,27 @@ export const createToken = (
 
 // TODO: have to go through setup because we cannot create a user w/ a password via the user API
 export const setupUser = (): Cypress.Chainable<Cypress.Response<any>> => {
-  return cy.request({
-    method: 'GET',
-    url: '/debug/provision',
-  })
+  const defaultUser = Cypress.env('defaultUser')
+  const userParam = defaultUser ? `?user=${defaultUser}` : ''
```
Comment (genehynson) [inline cypress/support/commands.ts:579]: the first time, defaultUser will be empty. Every time after that, it will be populated

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 59 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2533)

File: cypress/e2e/cloud/about.test.ts [hunk 2]
```diff
@@ -2,25 +2,25 @@ import {Organization} from '../../../src/types'
...
-      cy.get('@org').then(({id}: Organization) => {
-        cy.setFeatureFlags({
-          uiUnificationFlag: true,
-        }).then(() => {
-          cy.quartzProvision({
-            accountType: 'free',
-            hasUsers: false,
+  beforeEach(() =>
+    cy.flush().then(() =>
+      cy.signin().then(() => {
```
Comment (genehynson) [inline cypress/e2e/cloud/about.test.ts:7]: we must wait on the async `cy.flush()` command to complete before signing in. Had to change this in every beforeEach

File: .circleci/config.yml [hunk 3]
```diff
@@ -136,7 +136,7 @@ jobs:
     steps:
       - run:
           name: Run monitor-ci tests
-          command: API_KEY=${MONITOR_CI_API_KEY} PULL_REQUEST=${CIRCLE_PULL_REQUEST} MONITOR_CI_BRANCH="master" UI_BRANCH=${CIRCLE_BRANCH} SHA=${CIRCLE_SHA1} /run-monitor-ci-tests.bash
+          command: API_KEY=${MONITOR_CI_API_KEY} PULL_REQUEST=${CIRCLE_PULL_REQUEST} MONITOR_CI_BRANCH="genehynson/update-provisioner-oss" UI_BRANCH=${CIRCLE_BRANCH} SHA=${CIRCLE_SHA1} /run-monitor-ci-tests.bash
```
Comment (genehynson) [inline .circleci/config.yml:139]: will undo this change before merging

**Comment by genehynson** [comment]
the oss tests passed over in the UI Pr so we're good to merge this
